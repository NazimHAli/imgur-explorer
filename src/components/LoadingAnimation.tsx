import "@/styles/component/loading-animation.scss";

const LoadingAnimation = () => {
  return (
    <div className="loading-container">
      <div className="mosaic-loader">
        <div className="cell d-0"></div>
        <div className="cell d-1"></div>
        <div className="cell d-2"></div>
        <div className="cell d-3"></div>
        <div className="cell d-1"></div>
        <div className="cell d-2"></div>
        <div className="cell d-3"></div>
        <div className="cell d-4"></div>
        <div className="cell d-2"></div>
        <div className="cell d-3"></div>
        <div className="cell d-4"></div>
        <div className="cell d-5"></div>
        <div className="cell d-3"></div>
        <div className="cell d-4"></div>
        <div className="cell d-5"></div>
        <div className="cell d-6"></div>
      </div>
    </div>
  );
};

export default LoadingAnimation;
