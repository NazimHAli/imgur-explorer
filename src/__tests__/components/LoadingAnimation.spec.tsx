import LoadingAnimation from "@/components/LoadingAnimation";

import { render } from "../fixtures/test-utils";

describe("LoadingAnimation", () => {
  test("loads", () => {
    const el = render(<LoadingAnimation />);
    expect(el.container).toMatchSnapshot();
  });
});
