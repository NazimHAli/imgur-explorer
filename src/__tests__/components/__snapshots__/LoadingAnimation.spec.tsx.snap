// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAnimation loads 1`] = `
<div>
  <div
    class="loading-container"
  >
    <div
      class="mosaic-loader"
    >
      <div
        class="cell d-0"
      />
      <div
        class="cell d-1"
      />
      <div
        class="cell d-2"
      />
      <div
        class="cell d-3"
      />
      <div
        class="cell d-1"
      />
      <div
        class="cell d-2"
      />
      <div
        class="cell d-3"
      />
      <div
        class="cell d-4"
      />
      <div
        class="cell d-2"
      />
      <div
        class="cell d-3"
      />
      <div
        class="cell d-4"
      />
      <div
        class="cell d-5"
      />
      <div
        class="cell d-3"
      />
      <div
        class="cell d-4"
      />
      <div
        class="cell d-5"
      />
      <div
        class="cell d-6"
      />
    </div>
  </div>
</div>
`;
